src/Core/Indexes/QuadTree.cs

```diff
@@ -0,0 +1,400 @@
+﻿namespace AEGIS.Indexes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class QuadTree : ISpatialIndex
+    {
+        protected class QuadTreeNode
+        {
+            /// <summary>
+            /// The region of the node.
+            /// </summary>
+            private Envelope bound;
+
+            /// <summary>
+            /// The children nodes of this node. Each internal node must always have 4 children.
+            /// </summary>
+            private List<QuadTreeNode> children = new List<QuadTreeNode>(4);
```

Comment:
Field initialization should be performed in a constructor. Also, if the field is not replaced after the constructor, using the `readonly` modifier makes the code more efficient.